JavaScript/packages/recognizers-date-time/src/dateTime/baseDatePeriod.ts

```diff
@@ -892,6 +896,52 @@ export class BaseDatePeriodParser implements IDateTimeParser {
         return result;
     }
 
+    // Resolve month overflow
+    private createDate(year: number, month: number, day: number): Date {
```

Comment:
Instead of doing it this way, create another method in DateUtil that safely adds the period you want to a date.